interpreter/per-host/src/main/scala/io/buoyant/transformer/perHost/LocalhostTransformer.scala

```diff
@@ -0,0 +1,43 @@
+package io.buoyant.transformer.perHost
+
+import com.twitter.finagle.Name.Bound
+import com.twitter.finagle.{Addr, Address, Name, NameTree}
+import com.twitter.util.Activity
+import io.buoyant.namer.{DelegateTree, DelegatingNameTreeTransformer}
+import java.net.InetAddress
+
+/**
+  * The localhost transformer filters the list of addresses down to only
+  * addresses that have the same IP address as localhost.  This can be used by
+  * an incoming router to only route traffic to local destinations.
+  */
+class LocalhostTransformer extends DelegatingNameTreeTransformer {
+
+  private[this] val localhost = InetAddress.getLocalHost.getAddress
```

Comment:
It's probably worth documenting explicitly how "local" is determined.  From the docs:

> Returns the address of the local host. This is achieved by retrieving the name of the host from the system, then resolving that name into an InetAddress.

It's unclear what this line does:
- when the host has no IP
- when the host's name isn't resolvable
- when the host has multiple IPs

Furthermore, in some cases (kubernetes), we may need to do additional work to determine locally-routable IPs.

I think I'd prefer that the class be configured with a `Seq[InetAddress]` (there can be multiple local addresses), and allow this to be configured.  Our default logic should probably fall back to getLocalHost.getAddress.

Perhaps this class should take an `isLocal: InetAddress => Boolean` argument so that this behavior may be easily swapped?